### PR TITLE
Elf add sym

### PR DESCRIPTION
**Pull request by serpilliere**

### CODE REVIEW

File: miasm2/core/locationdb.py [hunk 1]
```diff
@@ -252,7 +252,7 @@ def add_location(self, name=None, offset=None, strict=True):
             # Non-strict mode
             if name_loc_key is not None:
                 known_offset = self.get_offset_location(name_loc_key)
-                if known_offset != offset:
+                if known_offset is not None and known_offset != offset:
```
Comment (commial) [inline miasm2/core/locationdb.py]: IMHO, this is not enough: if `known_offset` is actually `None`, the returned `name_loc_key` will not have its offset set, so there is a set offset to do.

File: miasm2/analysis/binary.py [hunk 2]
```diff
@@ -209,6 +209,16 @@ def parse(self, data, vm=None, addr=0, apply_reloc=False, **kwargs):
             # Add known symbols (vm_load_elf already does it)
             fill_loc_db_with_symbols(self._executable, self.loc_db, addr)
 
+            if hasattr(self._executable.sh, 'dynsym'):
```
Comment (commial) [inline miasm2/analysis/binary.py]: This should rather be done in `fill_loc_db_with_symbols`

Comment (serpilliere) [reply miasm2/analysis/binary.py]: I agree with you: I moved the code

File: miasm2/core/locationdb.py (hunk 1 above)
Comment (serpilliere) [reply miasm2/core/locationdb.py]: Yes! I added the set_offset.

Comment (serpilliere) [reply miasm2/core/locationdb.py]: I agree with you: I moved the code

Comment (serpilliere) [reply miasm2/core/locationdb.py]: Yes! I added the set_offset.

File: miasm2/analysis/binary.py (hunk 2 above)
Comment (serpilliere) [reply miasm2/analysis/binary.py]: I agree with you: I moved the code

**Pull request merged**